### PR TITLE
Add possibility to update text and fontsize

### PR DESCRIPTION
**Pull request by davidbonan**
Add a generic component ElementOption who simple provide an label/input
Add handler and render for update text/fontsize and re-measured content

### CODE REVIEW

**Comment by vjeux** [comment]
@davidbonan thanks for contributing!

@paulomenezes since you implemented the rest of options, could you review this one too?

File: src/components/ElementOption.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    props.onChange(e.currentTarget.value);
+  };
+  return (
+    <>
+      <label style={{ display: "flex", flexDirection: "row" }}>
+        <span style={{ marginRight: 5, fontSize: 14, minWidth: 40 }}>
+          {props.label}:
+        </span>
+        <input
+          onChange={handleChangeInput}
```
Comment (paulomenezes) [inline src/components/ElementOption.tsx]: The size input can be a number type, to increase with arrow keys.

File: src/index.tsx [hunk 2]
```diff
@@ -1465,6 +1522,12 @@ class App extends React.Component<{}, AppState> {
           </div>
           {someElementIsSelected() && (
             <>
+              <h4>Element's options</h4>
+              <div className="panelColumn">
+                {elementSelectedOptions.length > 0
+                  ? elementSelectedOptions
+                  : "No options available"}
```
Comment (paulomenezes) [inline src/index.tsx]: I think is better hide the "Element's options" then show the "No options available". Only show if there is a text element selected

**Comment by paulomenezes** [comment]
Some improvements:
- It's not currently working when select two texts, when edit the text or the size, only update one element, maybe hide the option when more than one element is selected or update all the selected texts.
- Both input can be empty, maybe some treatment in this case? 
- The property should be grouped with another element properties, but I'll open a PR to move the other ones to top.

**Comment by Fausto95** [comment]
Sincd we're updating text properties, we could probably give the ability to update as well the font type!

File: src/index.tsx [hunk 3]
```diff
@@ -1392,9 +1388,70 @@ class App extends React.Component<{}, AppState> {
     this.setState({ currentItemBackgroundColor: color });
   };
 
+  private updateElement = (element: any, values: any) => {
+    /** Use of Object.assign instead spread because we want to keep ref to element for update is value in initial object */
+    Object.assign(element, values);
```
Comment (nanot1m) [inline src/index.tsx]: Have you tested undo action? Mutating original element can be dangerous here

Comment (davidbonan) [reply src/index.tsx]: Undo action exist ? Cmd+Z ? I don’t see it

**Comment by davidbonan** [comment]
> Sincd we're updating text properties, we could probably give the ability to update as well the font type!

Font familly added right now

File: src/index.tsx (hunk 2 above)
Comment (davidbonan) [reply src/index.tsx]: @paulomenezes done

File: src/components/ElementOption.tsx (hunk 1 above)
Comment (davidbonan) [reply src/components/ElementOption.tsx]: @paulomenezes Done => replace with select box

**Comment by davidbonan** [comment]
> Some improvements:
> 
> * It's not currently working when select two texts, when edit the text or the size, only update one element, maybe hide the option when more than one element is selected or update all the selected texts.
=> Done, the value input was hide on selected multiple text element but not the fontSize and fontFamilly

> * Both input can be empty, maybe some treatment in this case?
=> So it's okay now for the font size/familly, but no for the value. 

> * The property should be grouped with another element properties, but I'll open a PR to move the other ones to top.
=> Hum okay so, i wait it ?

File: src/index.tsx (hunk 3 above)
Comment (davidbonan) [reply src/index.tsx]: Oh my PR was launch before your undo/redo was merge, sorry. I rebase on master

**Comment by davidbonan** [comment]
=> Rebase to master
=> Fix integration with "Selection" panel
=> test undo/redo => ok

File: src/index.tsx [hunk 4]
```diff
@@ -1397,9 +1395,97 @@ class App extends React.Component<{}, AppState> {
...
+              value={selectedTextElements[0].text}
+              onChange={(text: string) =>
+                this.changeTextElement({ text })
+              }
+            />
+          );
+        }        
+        // Font size option
+        elementSelectedTextOptions.push(
+          <ElementOption
```
Comment (vjeux) [inline src/index.tsx]: Not sure if you've seen, but there's a <ButtonSelect> component in the codebase

File: src/index.tsx [hunk 5]
```diff
@@ -1469,96 +1555,100 @@ class App extends React.Component<{}, AppState> {
...
+              
+              <div className="panelColumn">
+                <h4>Selection</h4>
+                <div className="buttonList">
+                  <button onClick={this.moveOneRight}>Bring forward</button>
+                  <button onClick={this.moveAllRight}>Bring to front</button>
+                  <button onClick={this.moveOneLeft}>Send backward</button>
+                  <button onClick={this.moveAllLeft}>Send to back</button>
+                </div>
+                {elementSelectedTextOptions.length > 0 && elementSelectedTextOptions}
```
Comment (vjeux) [inline src/index.tsx]: This is off, the `}` shouldn't end at this line.

**Comment by davidbonan** [comment]
I closed that PR because the diff on master is too big and it's simpler to open new branch.

New PR on #278

**Pull request closed without merging**